### PR TITLE
Fix overridable option fields.

### PR DESCRIPTION
**Pull request by jacarey**
Added a test and some java reflection to set all hidden fields to the child value.

### CODE REVIEW

File: src/java/picard/cmdline/CommandLineParser.java [hunk 1]
```diff
@@ -667,7 +667,14 @@ private boolean parseOption(String key, final String stringValue, final boolean
                 optionDefinition.hasBeenSet = true;
                 optionDefinition.hasBeenSetFromOptionsFile = optionsFile;
             } else {
-                optionDefinition.field.set(callerOptions, value);
+                //get all fields with this name and set them to the argument.
+                final String fieldName = optionDefinition.field.getName();
+                final Field[] fields = callerOptions.getClass().getFields();
+                for(final Field field : fields){
+                    if(field.getName().equals(fieldName)){
```
Comment (nh13) [inline src/java/picard/cmdline/CommandLineParser.java]: ```if (```

File: src/java/picard/cmdline/CommandLineParser.java [hunk 2]
```diff
@@ -667,7 +667,14 @@ private boolean parseOption(String key, final String stringValue, final boolean
                 optionDefinition.hasBeenSet = true;
                 optionDefinition.hasBeenSetFromOptionsFile = optionsFile;
             } else {
-                optionDefinition.field.set(callerOptions, value);
+                //get all fields with this name and set them to the argument.
+                final String fieldName = optionDefinition.field.getName();
+                final Field[] fields = callerOptions.getClass().getFields();
+                for(final Field field : fields){
```
Comment (nh13) [inline src/java/picard/cmdline/CommandLineParser.java]: ```for (```

File: src/java/picard/cmdline/CommandLineParser.java [hunk 3]
```diff
@@ -913,6 +920,10 @@ private void handleOptionAnnotation(final Field field) {
                 optionDefinitions.add(optionDefinition);
                 optionMap.put(optionDefinition.name, optionDefinition);
             }
+            //we are overridable but we already exist in the map so we need to update the hidden field value
+            else if(optionMap.containsKey(optionDefinition.name)){
```
Comment (nh13) [inline src/java/picard/cmdline/CommandLineParser.java]: spaces before/after brackets

File: src/tests/java/picard/cmdline/CommandLineParserTest.java [hunk 4]
```diff
@@ -935,4 +935,44 @@ public void testStaticPropagationNegative() {
...
+
+    }
+
+    class OverridePropagation extends StaticParent {
+        @Option
+        public String STRING3 = "String3Overriden";
+    }
+
+    @Test
+    public void testOveriddenOptions(){
```
Comment (nh13) [inline src/tests/java/picard/cmdline/CommandLineParserTest.java]: ```() {```

File: src/tests/java/picard/cmdline/CommandLineParserTest.java [hunk 5]
```diff
@@ -935,4 +935,44 @@ public void testStaticPropagationNegative() {
...
+        @Option
+        public String STRING1 = "String1ParentDefault";
+
+        @Option
+        public String STRING2 = "String2ParentDefault";
+
+        @Option(overridable = true)
+        public String STRING3 = "String3ParentDefault";
+
+        public void doSomething(){
```
Comment (nh13) [inline src/tests/java/picard/cmdline/CommandLineParserTest.java]: ```() {```

**Comment by nh13** [comment]
:+1:

**Comment by jacarey** [comment]
@yfarjoun You are good to go to rebase your branch.

**Comment by yfarjoun** [comment]
sweet!

**Pull request merged**